packages/xpetra/sup/Utils/Xpetra_TripleMatrixMultiply.hpp

```diff
@@ -147,6 +147,30 @@ namespace Xpetra {
                         fillParams);
       }
 
+      // transfer striding information
+      RCP<const Map> domainMap = Teuchos::null;
+      RCP<const Map> rangeMap  = Teuchos::null;
+
+      const std::string viewLabel("stridedMaps");
+      const size_t        blkSize = 1;
+      std::vector<size_t> stridingInfo(1, blkSize);
+      LocalOrdinal        stridedBlockId = -1;
+      
+      if (R->IsView(viewLabel)) {
```

Comment:
Why not simplify this `if-else` as
```c++
auto rangeMap  = transposeR ? R.getColMap(viewLabel) : R.getRowMap(viewLabel);
if (R->IsView(viewLabel))
  rangeMap  = Xpetra::StridedMapFactory<LocalOrdinal, GlobalOrdinal, Node>::Build(rangeMap,  stridingInfo, stridedBlockId);
```
Also, maybe use `stridedViewLabel` instead of generic `viewLabel`?

Reply:
Wouldn't this bug out on the first line when R does not have `(R->IsView(viewLabel))`?

Reply:
oh, stupid me! I misread the code, and missed `viewLabel` in `getColMap`. Please disregard all my comments!